src/utils/WindowPreviewer.js

```diff
@@ -0,0 +1,42 @@
+import { $ } from './dom'
+
+class WindowPreviewer {
+  constructor () {
+    this.previewerWindow = null
+    this.iframeStyles = {
+      background: '#fff',
+      border: '0',
+      height: '100%',
+      width: '100%'
+    }
+  }
+
+  setupWindowIframe () {
+    this.previewerWindow.document.body.style.margin = 0
+    this.iframe = this.previewerWindow.document.createElement('iframe')
+    Object.entries(this.iframeStyles).forEach(
+      ([attr, val]) => (this.iframe.style[attr] = val)
+    )
+    this.previewerWindow.document.body.appendChild(this.iframe)
+  }
+
+  updateWindowContent (html) {
+    if (this.previewerWindow) {
+      this.iframe.setAttribute(
+        'srcdoc',
+        html || $('iframe').getAttribute('srcdoc')
+      )
+    }
+  }
```

Comment:
this is guard clause and help us to have code more readable 

```suggestion
  updateWindowContent (html) {
    if (!this.previewerWindow) return
    
    this.iframe.setAttribute(
      'srcdoc',
      html || $('iframe').getAttribute('srcdoc')
    )
  }
```